src/schema.ts

```diff
@@ -128,47 +174,63 @@ export function build(data: any, opt: QueryConfig = {}): Schema {
     }
   }
 
-  return new Schema(fieldSchemas);
+  const derivedTableSchema: TableSchema<FieldSchema> = {
```

Comment:
```
const derivedTableSchema: TableSchema<FieldSchema> = {
   ...tableSchema,
   fields: fieldSchemas
};
```
would be shorter and also support future additional properties in the table schema?